src/grokcore/annotation/tests/annotation/lazy.py

```diff
@@ -43,6 +43,21 @@
   >>> str(grok.queryAnnotation(manfred, ILazy).lazy_attribute)
   'We have a value!'
 
+One might expect to be able to explicitely mark the annotation as modified
+(for example when the property is a sequence or mapping that is modified).
+Flagging the annotation as modiffied should be properly delegated to the
+underlying persisten storage object:
+
+  >>> annotation_object = grok.queryAnnotation(manfred, ILazy)
+  >>> annotation_object.storage is not None
+  True
+
+  >>> annotation_object._p_changed
+  False
+
+# XXX if someone knows a way to reliably test setting the _p_changed and
+# reading it back in this test let me know...
+
```

Comment:
```suggestion
  >>> from unittest import mock
  >>> annotation_object.storage._p_jar = mock.Mock()

  >>> annotation_object._p_changed
  False

  >>> annotation_object._p_changed = True

  >>> annotation_object._p_changed
  True
```

I don't really know what's going on and if this helps, but if the storage doesn't have a _p_jar the _p_changed getter will always return `False` https://github.com/zopefoundation/persistent/blob/61d6374c10ecc95b8660bc2a45b21b5c57fbd236/persistent/persistence.py#L177

Reply:
The solution: https://github.com/zopefoundation/persistent/blob/61d6374c10ecc95b8660bc2a45b21b5c57fbd236/persistent/tests/utils.py#L2

Thanks @Recursing for the hint in the right direction.